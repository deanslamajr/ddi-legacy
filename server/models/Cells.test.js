const {createNewCell} = require('./Cells');
const {sequelize, __setFindOne} = require('../adapters/db');

const {create, findOne} = sequelize.define();


jest.mock('shortid');
jest.mock('../adapters/db');

describe('models/Cells', () => {
  const comicId = 'comicId';
  const userId = 'userId';

  beforeEach(() => {
    create.mockClear();
    findOne.mockClear();
  })

  describe('createNewCell', () => {
    it('should create a new Cell with a unique filename, url_id', async () => {
      const mockedExistingCells = [
        'some', 'existing', 'cells'
      ];

      // set cell.create mock
      mockedExistingCells.forEach(mockCell => __setFindOne(mockCell));


      // invoke createNewCell
      await createNewCell({});

      // 3 calls against existing cells, either
      //   * Cells.findOne({ where: { url_id: urlId }})
      //   * Cells.findOne({ where: { image_url: filename }})
      // 2 calls against non existing cells
      // = 5 calls total
      expect(findOne).toHaveBeenCalledTimes(5);
    });

    it('should create the new Cell with the given comicId, userId', async () => {
      await createNewCell({comicId, userId});
      expect(create.mock.calls[0][0]).toMatchSnapshot();
    });

    it('should return the filename and urlId', async () => {
      const response = await createNewCell({comicId, userId});
      expect(response).toMatchSnapshot();
    });
  });
});