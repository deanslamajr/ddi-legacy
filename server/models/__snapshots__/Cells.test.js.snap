// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/Cells createNewCell should create the new Cell with the given comicId, userId 1`] = `
Object {
  "comic_id": "comicId",
  "creator_user_id": "userId",
  "image_url": "mockGeneratdId.png",
  "url_id": "mockGeneratdId",
}
`;

exports[`models/Cells createNewCell should return the filename and urlId 1`] = `
Object {
  "filename": "mockGeneratdId.png",
  "urlId": "mockGeneratdId",
}
`;
